### ISSUE TITLE
RFE: switch from jansson to json-c

### ISSUE THREAD

**Issue by kloczek** [universal-ctags/ctags#2895]
On my minimal build system ctags is requires by source-highlight which is used by gdb.
I just foind that ctags is using jansson.
It would be good IMO to swich to json-c as more popular JSON library or provide option to build against exact json library.
Looking on fedora packages repository:
```
[root@barrel ~]# dnf repoquery --whatrequires jansson --disablerepo g2v | wc -l
Last metadata expiration check: 2:34:52 ago on Mon 01 Mar 2021 12:37:36 GMT.
68
[root@barrel ~]# dnf repoquery --whatrequires json-c --disablerepo g2v | wc -l
Last metadata expiration check: 2:35:03 ago on Mon 01 Mar 2021 12:37:36 GMT.
120
````

**Comment by masatake** [comment]
In your use case, do you need the JSON output?
If not, you can just disable the JSON output feature with `./configure --disable-json`. So you don't need jansson.
I guess source-highlight may use tags output only.

BTW, I found both libraries are available in RHEL8.

**Comment by kloczek** [comment]
That is another option :)

Nevertheless in mean time I made small analyse/investigation and I found that I wrong and looks like jansson is a bit better JSON librarry than json-c,
https://github.com/miloyip/nativejson-benchmark
So .. I think that I should spend some time to investigate possible migration some packages from use json-c or json-glib (currently most frequently used JSON library but way worse in context of executable code size and performance.
```
# size /usr/lib64/libjson-c.so.5.1.0 /usr/lib64/libjansson.so.4.13.0 /usr/lib64/libjson-glib-1.0.so.0.600.2
   text    data     bss     dec     hex filename
  63598    2616     312   66526   103de /usr/lib64/libjson-c.so.5.1.0
  52007     984      16   53007    cf0f /usr/lib64/libjansson.so.4.13.0
 163096    3416     416  166928   28c10 /usr/lib64/libjson-glib-1.0.so.0.600.2
```
On top of raw code size json-glib is using gobject, gio and glib:
```
$ objdump -x /usr/lib64/libjson-glib-1.0.so.0.600.2 | grep NEEDED
  NEEDED               libgio-2.0.so.0
  NEEDED               libgobject-2.0.so.0
  NEEDED               libglib-2.0.so.0
  NEEDED               libc.so.6
```
In other words feel free to close this ticket :P

**Comment by masatake** [comment]
I will accept the code adding to support json-c to ctags.
I will help people try to add to support json-c to ctags.
However, I myself not work on it directly.
